libs/block-editor/src/lib/demo/block-editor.component.ts

```diff
@@ -42,21 +56,116 @@ export class BlockEditorComponent implements OnInit {
+        this.tippy = tippy(this.editor.view.dom, {
+            duration: 150,
+            getReferenceClientRect: null,
+            content: this.componentLinkForm.location.nativeElement,
+            interactive: true,
+            trigger: 'manual',
+            placement: 'bottom',
+            hideOnClick: 'toggle',
+            animation: 'fade'
+        });
+    }
+
+    initLinkFormComponent(injector: Injector, resolver: ComponentFactoryResolver): ComponentRef<BubbleMenuLinkFormComponent> {
+        const factory = resolver.resolveComponentFactory(BubbleMenuLinkFormComponent);
+        const component = factory.create(injector);
+
+        // Attach Events
+        component.instance.hideForm.subscribe(() => this.hideLinkForm());
+        component.instance.removeLink.subscribe(() => this.removeLink());
+        component.instance.setLink.subscribe((event) => this.setLink(event));
+        component.instance.toggleHighlight.subscribe((event) => this.toggleHighlight(event));
+        
+        component.changeDetectorRef.detectChanges();
+        return component;
+    }
+
+    toggleLinkForm() {
+
+        if(!this.componentLinkForm) {
+            this.componentLinkForm = this.initLinkFormComponent(this.injector, this.resolver);
+            this.createTippy();
+        }
+
+        if( this.tippy.state.isShown ) {
+            this.hideLinkForm();
+        } else {
+            this.tippy.show();
+            this.setTippyPosition();
+            this.setInputLink();
+            this.focusInputLink();
+        }
+        this.detectLinkFormChanges();
+    }
+
+    hideLinkForm() {
+        this.editor.view.focus();
+        this.tippy.hide();
+        this.toggleHighlight(false);
+    }
+
+    setTippyPosition() {
+        const { view } = this.editor;
+        const { selection } = this.editor.state;
+        this.tippy.setProps({
+            getReferenceClientRect: () => posToDOMRect(view, selection.from, selection.to)
+        });
+    }
+
+    setLink( link: string ) {
+        if( link ) {
+            this.editor.commands.setLink({ href: link });
+        }
+        this.hideLinkForm();
+    }
+
+    removeLink() {
+        this.editor.commands.unsetLink();
+        this.hideLinkForm();        
+    }
+
+    toggleHighlight( add: boolean ) {
+        if ( add ) {
+            this.editor.commands.setHighlight();
+        } else {
+            this.editor.commands.unsetHighlight();
+        }
+    }
+
+    setInputLink() {
+        this.componentLinkForm.instance.nodeLink = this.getNodeLink();
+        this.componentLinkForm.instance.newLink = this.getNodeLink();
+    }
+
+    detectLinkFormChanges() {
+        this.componentLinkForm.changeDetectorRef.detectChanges();
+    }
+
+    focusInputLink() {
+        this.componentLinkForm.instance.focusInput();
+    }
+
+    private getNodeLink(): string {
+        return this.editor.isActive('link')
+            ? this.editor.getAttributes('link').href
+            : '';
```

Comment:
All this code should not be here, this should be an extension or something, not in the block editor, we need to check this.